phac-nml/irida-next
# Enhancement: Samples table pagination and page size

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Enhancements to the samples tables including:

1. Page size (limit) selector
2. Advanced pagination

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/user-attachments/assets/00a97cf1-c28b-439a-bd12-35afede109de)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Go to http://localhost:3000/-/groups/streptococcus/-/samples
2. New components should be on the bottom of the page
3. Clicking on the page size and selecting a different page size should update the table with the new size, always starting back on the first page.
4. Clicking around on the pagination component should take you to that page in the table.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
